t/api/02-iso.t

```diff
@@ -966,4 +965,46 @@ subtest 'PUBLISH and STORE variables cannot include slashes' => sub {
     $schema->txn_rollback;
 };
 
+subtest 'Expand specified variables when scheduling iso' => sub {
+    $schema->txn_begin;
+    add_opensuse_test(
+        'test',
+        BUILD_HA            => '%BUILD%',
+        BUILD_SDK           => '%BUILD_HA%',
+        SHUTDOWN_NEEDS_AUTH => 1,
+        HDD_1   => '%DISTRI%-%VERSION%-%ARCH%-%BUILD%@%MACHINE%-minimal_with_sdk%BUILD_SDK%_installed.qcow2',
+        MACHINE => ['32bit', '64bit'],
+    );
+    my $res = schedule_iso({%iso, _GROUP_ID => '1002', TEST => 'test', BUILD => '176.6'}, 200);
+    is($res->json->{count}, 2, 'two job templates were scheduled');
+    $res = schedule_iso({%iso, _GROUP_ID => '1002', TEST => 'test', BUILD => '176.6', MACHINE => '64bit'}, 200);
+    is($res->json->{count}, 1, 'only the job template which machine is 64bit was scheduled');
+    my $result = $jobs->find($res->json->{ids}->[0])->settings_hash;
+    delete $result->{NAME};
+    is_deeply(
+        $result,
+        {
+            BUILD               => '176.6',
+            BUILD_HA            => '176.6',
+            BUILD_SDK           => '176.6',
+            HDD_1               => 'opensuse-13.1-i586-176.6@64bit-minimal_with_sdk176.6_installed.qcow2',
+            ISO_MAXSIZE         => '4700372992',
+            QEMUCPU             => 'qemu64',
+            VERSION             => '13.1',
+            BACKEND             => 'qemu',
+            WORKER_CLASS        => 'qemu_i586',
+            MACHINE             => '64bit',
+            DVD                 => '1',
+            FLAVOR              => 'DVD',
+            TEST_SUITE_NAME     => 'test',
+            ISO                 => 'openSUSE-13.1-DVD-i586-Build0091-Media.iso',
+            DISTRI              => 'opensuse',
+            SHUTDOWN_NEEDS_AUTH => '1',
+            TEST                => 'test',
+            ARCH                => 'i586'
+        },
+        'specified variables were expanded correctly'
+    );
```

Comment:
To keep this shorter I recommend you only check for relevant entries, e.g. check MACHINE

Reply:
If you do not mind, I prefer to check all the settings. Because I also added some variables such as `%BUILD_HA%`. In my opinion, it is not only used to check MACHINE, but also other variables that need to be replaced.

Reply:
The thing is: It should be clear in this test what settings we really care about. Being too specific is harmful because any next contributor that changes something so that the test fails does not understand easily which parts are relevant. Of course you can check for multiple values but not for all. I recommend to use the `json_is` test method here.

Reply:
I modified the code just checking few import variables.  If I understand correctly, we cannot use `json_is` here, because `schedule_iso` returns `return $t->tx->res`.